src/com/intellij/rt/coverage/data/LineData.java

```diff
@@ -27,6 +27,7 @@
 
   private int myId = -1;
   private int myHits = 0;
+  private int myInstructions = 0;
```

Comment:
It is better to save extra fields to some other place to decrease memory consumption when instructions coverage is disabled.

Reply:
It is hard to support merge operations when instructions are stored separately.